test/common/helpers.test.js

```diff
@@ -0,0 +1,101 @@
+import {
+  refPush,
+} from '../../common/script/index';
+
+import {
+  $w,
+  countExists,
+} from '../../common/script/helpers';
+
+import {
+  keys,
+  size,
+} from 'lodash';
+
+describe('convenience functions', () => {
+  describe('$w', () => {
+    it('converts a space-delimited string to an array', () => {
+      expect($w('space delimited string')).to.eql(['space', 'delimited', 'string']);
+    });
+  });
+
+  describe('refPush', () => {
+    let testObject = {};
+    const UUID = 'identity';
+    const ITEM_WITH_UUID = {id: UUID, data: 'abcdef'};
+    const UUID_2 = 'uniqueness';
+
+    it('adds an item to the given object', () => {
+      refPush(testObject, ITEM_WITH_UUID);
+      expect(size(testObject)).to.eql(1);
+    });
+
+    it('uses included UUID if supplied', () => {
+      expect(testObject[UUID]).to.exist;
+    });
+
+    it('includes properties of added item', () => {
+      expect(testObject[UUID].id).to.eql(ITEM_WITH_UUID.id);
+      expect(testObject[UUID].data).to.eql(ITEM_WITH_UUID.data);
+    });
+
+    it('gives first item a sort position of 0', () => {
+      expect(testObject[UUID].sort).to.eql(0);
+    });
+
+    it('appends additional items', () => {
+      refPush(testObject, {});
+      expect(size(testObject)).to.eql(2);
+    });
+
+    it('assigns a UUID to the newly added item if none was supplied', () => {
+      const GENERATED_UUID = keys(testObject)[1];
+      const UUID_LENGTH = 36;
+
+      expect(GENERATED_UUID.length).to.eql(UUID_LENGTH);
```

Comment:
if you want here to validate the UUID you can use the `validator` module `isUUID` method